packages/compass-aggregations/src/components/focus-mode/focus-mode-stage-editor.spec.tsx

```diff
@@ -6,18 +6,28 @@ import { Provider } from 'react-redux';
 
 import configureStore from '../../../test/configure-store';
 import { FocusModeStageEditor } from './focus-mode-stage-editor';
+import { ConnectionInfoProvider } from '@mongodb-js/connection-storage/provider';
```

Comment:
Can we think about how we can implement this without the need for every single test to be adjusted? In most cases they don't care about this, right?

Reply:
The tests themselves probably don't but its kinda needed because we started using the hook - useConnectionInfo and does end up somewhere in the render path.

Reply:
Yes, and that's the general limitation of using hooks with context, but this shouldn't prevent us from making our lives easier and making writing tests comfortable, it's part of this work to do that. You can check query bar or workspaces plugin if you need some inspiration how this can be approached, or we can chat abot it too if you need help. My main concern is that if writing a test for even the smallest component will require you to wrap it in 10 unrelated providers, nobody will be writing these tests or time to write the test will be spent unproductively doing unrelated setup

Reply:
That's also why I'm advocating for limiting the amount of data you make these things depend on to the required minimum: we really shouldn't make our lives harder with the patterns we introduce